js/components/portal-dashboard/feedback/rubric-table.tsx

```diff
@@ -0,0 +1,155 @@
+import React from "react";
+// import { Map } from "immutable";
+import Markdown from "markdown-to-jsx";
+import LaunchIcon from "../../../../img/svg-icons/launch-icon.svg";
+
+import css from "../../../../css/portal-dashboard/feedback/rubric-table.less";
+interface IProps {
+  activityStarted: boolean;
+  rubric: any;
+  student: any;
+  rubricFeedback: any;
+  activityId: string;
+  activityIndex: number;
+  updateActivityFeedback?: (activityId: string, activityIndex: number, platformStudentId: string, feedback: any) => void;
+}
+export class RubricTableContainer extends React.PureComponent<IProps> {
+  render() {
+    const { activityStarted, rubric } = this.props;
+    const { criteria } = rubric;
+
+    if (activityStarted) {
+      return (
+        <div className={css.rubricContainer} data-cy="rubric-table">
+          {this.renderColumnHeaders(rubric)}
+          <div className={css.rubricTable}>
+            {criteria.map((crit: any) =>
+              <div className={css.rubricTableRow} key={crit.id} id={crit.id}>
+                <div className={css.rubricDescription}>
+                  <Markdown>{crit.description}</Markdown>
+                </div>
+                {this.renderRatings(crit)}
+              </div>
+            )}
+          </div>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  private renderColumnHeaders = (rubric: any) => {
+    const { referenceURL } = rubric;
+    return (
+      <div className={css.columnHeaders}>
+        <div className={css.rubricDescriptionHeader}>
+          <div className={css.scoringGuideArea}>
+            <a className={css.launchButton}>
+              <LaunchIcon className={css.icon} href={referenceURL} target="_blank" />
+            </a>
+            Scoring Guide
+        </div>
+          <div className={css.rubricDescriptionTitle}>{rubric.criteriaLabel}</div>
+        </div>
+        { rubric.ratings.map((rating: any) =>
+          <div className={css.rubricScoreHeader} key={rating.id}>
+            <div className={css.rubricScoreLevel}>{rating.label}</div>
+            {rubric.scoreUsingPoints && <div className={css.rubricScoreNumber}>({rating.score})</div>}
+          </div>
+        )
+        }
+      </div>
+    );
+  }
+
+  private renderRatings = (crit: any) => {
+    const { rubric } = this.props;
+    const { ratings } = rubric;
```

Comment:
You don't need to pass `crit` into this function,  You can get it here:
`const { criteria, ratings } = rubric;`

Reply:
Crit is actually the mapped criteria, so would have to map criteria at each function if I use the props.